File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/models/AbstractUser.java
package com.example.myanimelistjava.models;

import java.util.Date;
import java.util.UUID;

public abstract class AbstractUser {
    public final UUID id;
    public String name;
    public String email;
    public String password;
    public final Date createDate;
    public final Date birthDate;

    public AbstractUser(String name, String email, String password, Date createDate, Date birthDate) {
        this.id = UUID.randomUUID();
        this.name = name;
        this.email = email;
        this.password = password;
        this.createDate = createDate;
        this.birthDate = birthDate;
    }

    public AbstractUser(UUID id, String name, String email, String password, Date createDate, Date birthDate) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.password = password;
        this.createDate = createDate;
        this.birthDate = birthDate;
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/models/AbstractUser.kt
package com.example.myanimelist.models

import java.time.LocalDate
import java.util.*

abstract class AbstractUser {
    abstract val id: UUID
    abstract var name: String
    abstract var email: String
    abstract var password: String
    abstract val createDate: LocalDate
    abstract val birthDate: LocalDate
    abstract val admin: Boolean
}
